pkg/watermark/fetch/offset_timeline_test.go

```diff
@@ -210,10 +211,52 @@ func TestOffsetTimeline_GetOffset(t1 *testing.T) {
 		},
 	}
 	for _, tt := range tests {
-		t1.Run(tt.name, func(t1 *testing.T) {
+		t.Run(tt.name, func(t1 *testing.T) {
 			if got := tt.args.timeline.GetOffset(tt.args.inputEventTime); got != tt.want {
 				t1.Errorf("GetOffset() = %v, want %v", got, tt.want)
 			}
 		})
 	}
 }
+
+func TestOffsetTimeline(t *testing.T) {
```

Comment:
This test case is to test how a new watermark is inserted based on the fixed logic above https://github.com/numaproj/numaflow/pull/520/files#diff-f1d5af08aee46bc6f1d1db8af46bd313aa1e909373b57c086d9b2aa6fb3809ceR101-R105